src/controller/cve.controller/cve.controller.js

```diff
@@ -1,170 +1,79 @@
 async function submitCve (req, res) {
   const cve = req.body
   const cveId = cve.CVE_data_meta.ID
-  const instance = axios.create({
-    timeout: 10000,
-    headers: {
-      'CVE-API-CNA': req.header('CVE-API-CNA'),
-      'CVE-API-SUBMITTER': req.header('CVE-API-SUBMITTER'),
-      'CVE-API-SECRET': req.header('CVE-API-SECRET')
-    }
-  })
-
-  try {
-    const cpsResult = await instance.post('https://cvecps-dev.mitre.org/cps/ws/cve/' + cveId + '/', cve)
-    const updatedCve = JSON.parse(cpsResult.data.Entry)
-
-    // CPS work item was approved
-    if (!cpsResult.data.Error) {
-      logger.info(cveId + ' has been created/updated in CPS')
-      logger.info(cpsResult.data.Message)
 
-      // Store CVE in MongoDB
+  // Find CVE in MongoDB
+  Cve.findOne()
+    .byCveId(cveId)
+    .exec((err, result) => {
+      if (err) {
+        logger.warn('An error ocurred in MongoDB while trying to check if ' + cveId + ' already exists.')
+        logger.error(err.stack)
+        return res.status(500).send('Internal Server Error')
+      }
+
+      if (result) {
+        logger.info(cveId + ' already exists.')
+        return res.status(400).json({ message: cveId + ' already exists.' })
+      }
+
+      // Create CVE in MongoDB if it doesn't exist
+      const newCve = new Cve({ cve: cve })
+
+      // Find and Update CVE in MongoDB
       Cve.findOneAndReplace()
         .byCveId(cveId)
-        .replaceOne({ cve: updatedCve })
+        .replaceOne(newCve)
         .setOptions({ upsert: true })
-        .exec((err, result) => {
+        .exec((err) => {
           if (err) {
-            logger.warn('An error ocurred in MongoDB. ' + cveId + ' was not created/updated in the database.')
+            logger.warn('An error ocurred in MongoDB. ' + cveId + ' was not created in the database.')
             logger.error(err.stack)
-          } else {
-            if (result.upserted) {
-              logger.info(cveId + ' was successfully created in MongoDB')
-            } else {
-              logger.info(cveId + ' was successfully updated in MongoDB')
-            }
+            return res.status(500).json({ message: 'Failed to create ' + cveId + '.' })
           }
-
-          // Respond to the user always
-          return res.status(200).json({ message: cveId + ' has been created' }) // send message to user
         })
-    } else {
-      logger.warn('The CPS work item was not approved. ' + cveId + ' was not created/updated.')
-      logger.warn(cpsResult.data.Error)
-      return res.status(500).json({ message: 'The CPS work item was not approved. ' + cveId + ' could not be created.' }) // send error message to user
-    }
-  } catch (err) {
-    // Request to CPS failed
-    logger.error(err.stack)
-    return res.status(500).send('Internal Server Error') // send error message to user
-  }
+      logger.info(cveId + ' has been created.')
+      return res.status(200).json({ message: cveId + ' has been created.' })
```

Comment:
This might execute before `Cve.findOneAndReplace()` finishes. Thus, put inside `Cve.findOneAndReplace()` if no error occurs.